### PR TITLE
Delete notifyexternals.*

### PR DESCRIPTION
**Pull request by jkotas**
Leftovers from loader lock MDA

### CODE REVIEW

File: src/coreclr/vm/i386/stublinkerx86.cpp [hunk 1]
```diff
@@ -3547,49 +3546,6 @@ VOID StubLinkerCPU::EmitDisable(CodeLabel *pForwardRef, BOOL fCallIn, X86Reg Thr
     // jnz RarePath
     X86EmitCondJump(pForwardRef, X86CondCode::kJNZ);
 
-#if defined(FEATURE_COMINTEROP) && !defined(FEATURE_CORESYSTEM)
```
Comment (jkotas) [inline src/coreclr/vm/i386/stublinkerx86.cpp:3550]: `FEATURE_CORESYSTEM` is not defined, so this was not getting compiled in.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/i386/stublinkerx86.cpp:3550]: Okay, but doesn't that mean it was being compiled in, the `defined` is negated?

Comment (jkotas) [reply src/coreclr/vm/i386/stublinkerx86.cpp:3550]: I meant to say `FEATURE_CORESYSTEM` is always defined (https://github.com/dotnet/runtime/blob/main/src/coreclr/clrdefinitions.cmake#L95). It means `!FEATURE_CORESYSTEM` is always false and this is never getting compiled in.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
@jkotas The failure is below. Perhaps this be related to the possible GC hole issue being tracked by @jkoritzinsky.

```
===========================================================================================================
/private/tmp/helix/working/A94209AB/w/B3F50A54/e /private/tmp/helix/working/A94209AB/w/B3F50A54/e
  Discovering: Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (found 496 test cases)
  Starting:    Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (parallel test collections = on, max threads = 4)

Assert failure(PID 32527 [0x00007f0f], Thread: 135455 [0x2111f]): m_alignpad == 0
    File: /Users/runner/work/1/s/src/coreclr/vm/syncblk.cpp Line: 2952
    Image: /private/tmp/helix/working/A94209AB/p/dotnet
```

**Comment by jkotas** [comment]
> Perhaps this be related to the possible GC hole issue being tracked

Yes, we have been seeing intermittent crashes like this in the CI recently, across multiple tests.

**Pull request merged**